gui/default/syncthing/core/syncthingController.js

```diff
@@ -2114,11 +2122,19 @@ angular.module('syncthing.core')
         }
 
         $scope.shareFolderWithDevice = function (folder, device) {
-            $scope.folders[folder].devices.push({
-                deviceID: device
-            });
-            $scope.config.folders = folderList($scope.folders);
-            $scope.saveConfig();
+            var folderCfg = $scope.folders[folder];
+            if (folderCfg.type == "receiveencrypted" || !$scope.pendingIsRemoteEncrypted(folder, device)) {
+                $scope.folders[folder].devices.push({
+                    deviceID: device
+                });
+                $scope.config.folders = folderList($scope.folders);
+                $scope.saveConfig();
+            } else {
+                // Open edit folder dialog to enter encryption password
+                $scope.editFolderExisting(folderCfg);
+                $scope.currentSharing.selected[device] = true;
+                $('.nav-tabs a[href="#folder-sharing"]').tab('show');
```

Comment:
This is not a beautiful coupling of presentation and logic...

Reply:
Made the tab to show on open a function parameter, such that this can be done in `editFolderModal`. Not sure if this act of beautification (if it is) isn't entirely lost in the overall ... non-beauty of `syncthingController.js` :)

Reply:
Yeah I don't think this makes it much better or worse really :)